postgresql/config.go

```diff
@@ -280,6 +284,25 @@ func (c *Client) Connect() (*DBConnection, error) {
 		var err error
 		if c.config.Scheme == "postgres" {
 			db, err = sql.Open(proxyDriverName, dsn)
+		} else if c.config.Scheme == "gcppostgres" && c.config.GCPIAMImpersonateServiceAccount != "" {
+			ts, err := impersonate.CredentialsTokenSource(context.Background(), impersonate.CredentialsConfig{
+				TargetPrincipal: c.config.GCPIAMImpersonateServiceAccount,
+				Scopes:          []string{"https://www.googleapis.com/auth/sqlservice.admin"},
+			})
+			if err != nil {
+				return nil, fmt.Errorf("Error creating token source with service account impersonation of %s: %w", c.config.GCPIAMImpersonateServiceAccount, err)
+			}
+			client, err := gcp.NewHTTPClient(gcp.DefaultTransport(), ts)
+			if err != nil {
+				return nil, fmt.Errorf("Error creating HTTP client with service account impersonation of %s: %w", c.config.GCPIAMImpersonateServiceAccount, err)
+			}
+			certSource := cloudsql.NewCertSourceWithIAM(client, ts)
+			opener := gcppostgres.URLOpener{CertSource: certSource}
+			dbURL, err := url.Parse(dsn)
+			if err != nil {
+				return nil, fmt.Errorf("Error parsing connection string: %w", err)
+			}
+			db, err = opener.OpenPostgresURL(context.Background(), dbURL)
```

Comment:
Thanks golangci-lint for catching that :muscle: , as you redefined a new `err` in the `if` scope line 288, this line will not defined the variable `err` we defined line 284 and this error will never been check.

The easy solution is to have a different name for the `err` you use between line 288 and 303 but this makes me think that we could probably have a dedicated small function to get the db connection for impersonate GCP so this would look like:
```golang
		} else if c.config.Scheme == "gcppostgres" && c.config.GCPIAMImpersonateServiceAccount != "" {
                    db, err = getGCPImpersonatedConnection(c.config.GCPIAMImpersonateServiceAccount)
                }
```